google-cloud-core/src/main/java/com/google/cloud/ServiceOptions.java

```diff
@@ -547,8 +545,12 @@ public static String getGoogApiClientLibName() {
   /**
    * Returns the library's version as a string.
    */
-  public static String getLibraryVersion() {
-    return LIBRARY_VERSION;
+  public static String getLibraryVersion(String artifactId) {
+    String version = getPomVersion(artifactId);
+    if (version == null) {
+      version = getManifestVersion();
```

Comment:
Should we try to get the manifest version of the artifact instead of google-cloud-core?